File: LeetCode/1.两数之和/Solution.kt
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        val map = mutableMapOf<Int,Int>()
        val result = IntArray(2)
        for (i in nums.indices){
            var j:Int? = map.get(target - nums[i])
            if (j != null){
                result.set(0,j)
                result.set(1,i)
                return result
            }
            map.put(nums[i],i)
        }
        return result
    }
}

File: LeetCode/1.两数之和/Solution.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        numDict = dict()
        for i,val in enumerate(nums):
            if target - val in numDict:
                return [numDict[target - val],i]
            numDict[val] = i
        return []